nixos/modules/system/boot/tmp.nix

```diff
@@ -30,6 +30,8 @@ with lib;
 
   config = {
 
+    systemd.additionalUpstreamSystemUnits = optional config.boot.tmpOnTmpfs "tmp.mount";
+
     systemd.mounts = mkIf config.boot.tmpOnTmpfs [
       {
         what = "tmpfs";
```

Comment:
We should probably also remove the unnecessary overrides (`what`, `where`) here since they're being provided by the upstream unit.

Reply:
If this PR is the way to go (and not https://github.com/NixOS/nixpkgs/pull/108860), can this be addressed?

Reply:
Done. `where` needs to be kept, as it's a mandatory option.

Reply:
(For clarification: this is because the unit name is derived from the `where`.)